### PR TITLE
Wassim branch

### PR DESCRIPTION
**Pull request by sakriwassim**

### CODE REVIEW

File: lib/jmap/account_id.g.dart [hunk 1]
```diff
@@ -6,11 +6,9 @@ part of 'account_id.dart';
 // JsonSerializableGenerator
 // **************************************************************************
 
-AccountId _$AccountIdFromJson(Map<String, dynamic> json) {
-  return AccountId(
-    const IdConverter().fromJson(json['id'] as String),
-  );
-}
+AccountId _$AccountIdFromJson(Map<String, dynamic> json) => AccountId(
```
Comment (dab246) [inline lib/jmap/account_id.g.dart:9]: Do not change the `format code` in JsonSerializableGenerator `.g` files without changing its properties.

File: lib/jmap/calendar/blob_found_not_parsable.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'blob_found_not_parsable.g.dart';
+
+@JsonSerializable()
+class BlobFoundNotParsable with EquatableMixin {
+  final String? accountId;
```
Comment (dab246) [inline lib/jmap/calendar/blob_found_not_parsable.dart:8]: It is recommended that `AccountId` replace the data type `String` to make the object more meaningful and easier to use later.

File: lib/jmap/calendar/blob_id.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import 'package:json_annotation/json_annotation.dart';
+
+part 'blob_id.g.dart';
+
+@JsonSerializable()
+class BlobId with EquatableMixin {
+  final String? uid;
+  final String? title;
+  final String? description;
+  final String? start;
```
Comment (chibenwa) [inline lib/jmap/calendar/blob_id.dart:17]: If possible favor using a stronger type (Date?)

File: lib/jmap/calendar/blob_id.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  final int? sequence;
+  final int? priority;
+  final String? freeBusyStatus;
+  final String? privacy;
+  final Organizer? organizer;
+  final List<Participants>? participants;
+  final ExtensionFields? extensionFields;
+  final List<RecurrenceRules>? recurrenceRules;
+
+  BlobId(
```
Comment (dab246) [inline lib/jmap/calendar/blob_id.dart:32]: Should write

```suggestion
BlobId({
  this.uid,
  this.title,
  this.description,
  this.start,
  this.duration,
  this.end,
  this.timeZone,
  this.location,
  this.method,
  this.sequence,
  this.priority,
  this.freeBusyStatus,
  this.privacy,
  this.organizer,
  this.participants,
  this.extensionFields,
  this.recurrenceRules
});
```

File: lib/jmap/calendar/blob_id.dart [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+import 'extension_fields.dart';
+import 'organizer.dart';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'blob_id.g.dart';
+
+@JsonSerializable()
+class BlobId with EquatableMixin {
```
Comment (chibenwa) [inline lib/jmap/calendar/blob_id.dart:13]: This is not a `BlobId` but likely a `CalendarEvent`

File: lib/jmap/calendar/extension_fields.dart [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'extension_fields.g.dart';
+
+@JsonSerializable()
+class ExtensionFields with EquatableMixin {
+  final List<String>? xOPENPAASVIDEOCONFERENCE;
+  final List<String>? xOPENPAASCUSTOMHEADER1;
```
Comment (chibenwa) [inline lib/jmap/calendar/extension_fields.dart:9]: The extensions fields can and should be arbitrary.


Type here needs to be `Map<String, List<String>>`

File: lib/jmap/calendar/blob_id.dart [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      this.participants,
+      this.extensionFields,
+      this.recurrenceRules});
+
+  factory BlobId.fromJson(Map<String, dynamic> json) => _$BlobIdFromJson(json);
+
+  Map<String, dynamic> toJson() => _$BlobIdToJson(this);
+
+  @override
+  List<Object?> get props => [
```
Comment (dab246) [inline lib/jmap/calendar/blob_id.dart:56]:
```suggestion
List<Object?> get props => [
  uid,
  title,
  description,
  start,
  duration,
  end,
  timeZone,
  location,
  method,
  sequence,
  priority,
  freeBusyStatus,
  privacy,
  organizer,
  participants,
  extensionFields,
  recurrenceRules
];
```

File: lib/jmap/calendar/extension_fields.dart [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'extension_fields.g.dart';
+
+@JsonSerializable()
+class ExtensionFields with EquatableMixin {
+  final List<String>? xOPENPAASVIDEOCONFERENCE;
```
Comment (dab246) [inline lib/jmap/calendar/extension_fields.dart:8]: Please give the attribute name meaning

File: lib/jmap/core/capability/mdn_capability.g.dart [hunk 9]
```diff
@@ -6,9 +6,8 @@ part of 'mdn_capability.dart';
 // JsonSerializableGenerator
 // **************************************************************************
 
-MdnCapability _$MdnCapabilityFromJson(Map<String, dynamic> json) {
-  return MdnCapability();
-}
+MdnCapability _$MdnCapabilityFromJson(Map<String, dynamic> json) =>
+    MdnCapability();
```
Comment (chibenwa) [inline lib/jmap/core/capability/mdn_capability.g.dart:10]: Unrelated change

File: test/jmap/email/get_list_email_test.dart [hunk 10]
```diff
@@ -261,5 +261,8 @@ void main() {
       expect(resultList?.list.length, equals(5));
       expect(resultList?.list, containsAll({expectMail1, expectMail2, expectMail3, expectMail4, expectMail5}));
     });
+ 
+ 
+
```
Comment (chibenwa) [inline test/jmap/email/get_list_email_test.dart:266]: Unrelated change

**Review by chibenwa — COMMENTED** [review]
Please revert changes in files unrelated to calendar.

**Review by dab246 — COMMENTED** [review]
Some coding conventions:
- Use `2 space ~ 1 tab` to separate the margin with each statement.
- If a function has many parameters, `each parameter` should be in `1 line`.
